### PR TITLE
Feature lif box 

### PR DESCRIPTION
**Pull request by theLamentingGirl**
Hi, I created a LIFBox class that wraps around the LIFBoxCell. Also created a test function for the module. Please review :)

### CODE REVIEW

**Review by Jegp — CHANGES_REQUESTED** [review]
Nice work! Could I ask you to look at two (minor) things?

1. You seem to have many changes in unrelated files. Which is not necessarily bad, but may indicate formatting issues. Could you run this by the [black](https://github.com/psf/black) formatter (`black .`) to just make sure everything is formatted neatly.
2. You also add a functional version of the lif_box (which is nice)! But could you kindly copy a test in for that as well? (in `norse/torch/functional/test`?) That will be super helpful for regressions.

File: .gitignore [hunk 1]
```diff
@@ -39,3 +39,4 @@ htmlcov/
 .venv/
 .secrets
 .coverage
+norse/torch/functional/rough.py
```
Comment (cpehle) [inline .gitignore:42]: where is this coming from

File: norse/torch/functional/encode.py [hunk 2]
```diff
@@ -134,6 +134,8 @@ def poisson_encode(
     seq_length: int,
     f_max: float = 100,
     dt: float = 0.001,
+    generator: torch.Generator = None,
```
Comment (cpehle) [inline norse/torch/functional/encode.py:137]: please open a separate pull request for this

File: norse/torch/functional/encode.py [hunk 3]
```diff
@@ -147,12 +149,15 @@ def poisson_encode(
...
+        generator (torch.Generator): Generator for pseudorandom numbers. Usually, generator.manual_seed(seed value) is passed as the argument
 
     Returns:
         A tensor with an extra dimension of size `seq_length` containing spikes (1) or no spikes (0).
     """
+    # if seed_generator is not None:
+
     return (
-        torch.rand(seq_length, *input_values.shape, device=input_values.device).float()
+        torch.rand(seq_length, *input_values.shape, device=input_values.device,generator = generator).float()
```
Comment (cpehle) [inline norse/torch/functional/encode.py:160]: same here

File: norse/torch/functional/encode.py [hunk 4]
```diff
@@ -161,6 +166,7 @@ def poisson_encode_step(
     input_values: torch.Tensor,
     f_max: float = 1000,
     dt: float = 0.001,
+    generator: torch.Generator = None,
```
Comment (cpehle) [inline norse/torch/functional/encode.py:169]: and here

File: norse/torch/functional/encode.py [hunk 5]
```diff
@@ -178,13 +184,14 @@ def poisson_encode_step(
...
-        torch.rand(*input_values.shape, device=input_values.device).float()
+        torch.rand(*input_values.shape, device=input_values.device,generator=generator).float()
         < dt * f_max * input_values
     ).float()
 
 
 def signed_poisson_encode(
-    input_values: torch.Tensor, seq_length: int, f_max: float = 100, dt: float = 0.001
+    input_values: torch.Tensor, seq_length: int, f_max: float = 100, dt: float = 0.001,
+    generator: torch.Generator = None,
```
Comment (cpehle) [inline norse/torch/functional/encode.py:194]: and here

File: norse/torch/functional/test/test_encode.py [hunk 6]
```diff
@@ -8,6 +8,7 @@
     constant_current_lif_encode,
     spike_latency_lif_encode,
     spike_latency_encode,
+    poisson_encode,
```
Comment (cpehle) [inline norse/torch/functional/test/test_encode.py:11]: and here

File: norse/torch/functional/test/test_encode.py [hunk 7]
```diff
@@ -129,3 +130,25 @@ def test_spike_latency_encode_without_batch_3():
         ]
     )
     assert torch.equal(actual, expected)
+
+def test_poisson_encode():
```
Comment (cpehle) [inline norse/torch/functional/test/test_encode.py:134]: not related

**Review by cpehle — CHANGES_REQUESTED** [review]
Can you please move all changes related to encoding and the associated tests to a separate pull request? They look good, but each pull request should correspond to a specific change.

**Comment by Jegp** [comment]
I'll close this for now. Happy to reopen later :)

**Pull request closed without merging**